packages/editor/src/client/atoms/emojiButton.tsx

```diff
@@ -11,54 +11,20 @@ import {Picker, BaseEmoji} from 'emoji-mart'
 export interface EmojiButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
   readonly icon: IconNames | SVGIcon
   readonly iconActive?: IconNames | SVGIcon
-  readonly active?: boolean
 }
 
-export const BaseEmojiButton = forwardRef<HTMLButtonElement, EmojiButtonProps>(
-  function BaseEmojiButton({icon, iconActive, active, ...props}, ref) {
-    icon = iconActive && active ? iconActive : icon
-    return (
-      <button
-        style={{
-          border: active ? 'blue 1px solid' : '',
-          fontSize: 16,
-
-          cursor: 'pointer',
-          borderRadius: 3,
-          backgroundColor: 'transparent',
-
-          padding: 2
-        }}
-        ref={ref}
-        {...props}>
-        <Icon icon={icon} element={icon} />
-      </button>
-    )
-  }
-)
-
 export function EmojiButton({icon, iconActive}: EmojiButtonProps) {
   const editor = useSlate()
-  const [isEmojiPickerOpen, setIsEmojiPickerOpen] = useState(false)
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false)
 
   const triggerRef = useRef<any>(null)
 
-  const open = () => {
-    if (triggerRef.current) {
-      triggerRef.current!.open()
-      // TODO rather be done with forwardRef ?
-      setIsEmojiPickerOpen(true)
-    }
-  }
-  const close = () => {
-    if (triggerRef.current) {
-      triggerRef.current!.close()
-      setIsEmojiPickerOpen(false)
-    }
-  }
+  const popOverRef = useCallback((node: any) => {
+    setIsPopoverOpen(!!node)
```

Comment:
Do the `!!` have a special role? Otherwise `!!node === node` wouldn't it?

Reply:
I actually found this on this repo. It turns any defined object into boolean, which is needed in this case.

Reply:
Ah ok, I didn't know that - just learned something new :)